File: tests/uitabview_test.cpp
#include "Monochrome.h"
using namespace mc;

int main()
{
	auto window = UIWindow::Create(WindowStyle::Modern, 1220, 760, "UITabView Test");
	window->SetBackgroundColor(Color(28, 21, 22, 1.0f));
	window->SetModernWindowButtonsColor(window->GetBackgroundColor());

	auto tabview = MakeRef<UITabView>(Frame(300, 80, 760, 600));
	tabview->AddTab("Element Properties")->layer.color = Color(80, 60, 10, 1);
	tabview->AddTab("Window Settings")->layer.color = Color(0, 80, 120, 1);
	tabview->AddTab("Config")->layer.color = Color(100, 40, 0, 1);
	tabview->AddTab("File 1")->layer.color = Color(0, 40, 20, 1);
	tabview->AddTab("File 2")->layer.color = Color(0, 20, 22, 1);
	tabview->StyleTabButtons([](Ref<UIButton>& tab) {
		tab->layer.color = Color(28, 28, 29, 1);
		tab->Label->color = Color::light_blue;
	});
	tabview->OpenTab("Window Settings");
	window->AddView(tabview);

	auto deleteButton = MakeRef<UIButton>();
	deleteButton->layer.frame.position += {100, 40};
	deleteButton->Label->Text = "Delete Tab";
	deleteButton->AddEventHandler<EventType::MouseButtonClicked>([tabview](Event& e, UIView* sender) -> bool {
		tabview->RemoveTab("Window Settings");
		return EVENT_HANDLED;
	});

	tabview->GetTab("Window Settings")->AddSubview(deleteButton);

	window->StartWindowLoop();
	return 0;
}

File: bindings/python/uitabview_test.py
from Monochrome import *

def style_tab_button(tab):
    tab.layer.color = Color(28, 28, 29)
    tab.label.color = Color(0, 80, 200)

def deleteButton_onclick(event, sender):
    global tabview
    tabview.remove_tab("Window Settings")

window = UIWindow(1220, 760, "UITabView Test")
window.set_background_color(Color(28, 21, 22))
window.set_modern_window_buttons_color(Color(28, 21, 22))

tabview = UITabView()
tabview.layer.frame.position = Point(300, 80)
tabview.layer.frame.size = Size(760, 600)
tabview.add_tab("Element Properties").layer.color = Color(80, 60, 10)
tabview.add_tab("Window Settings").layer.color = Color(0, 80, 120)
tabview.add_tab("Config").layer.color = Color(100, 40, 0)
tabview.add_tab("File 1").layer.color = Color(0, 40, 20)
tabview.add_tab("File 2").layer.color = Color(0, 20, 22)
tabview.style_tab_buttons(style_tab_button)
tabview.open_tab("Window Settings")
window.add_view(tabview)

deleteButton = UIButton()
deleteButton.layer.frame.position = Point(160, 80)
deleteButton.label.text = "Delete Tab"
deleteButton.add_mouse_clicked_event_handler(deleteButton_onclick)

tabview.get_tab("Window Settings").add_subview(deleteButton)

window.start_window_loop()
